### PR TITLE
De-duplicate code (Home/End navigation)

### PR DESCRIPTION
**Pull request by t2b3**

### CODE REVIEW

File: src/document.h [hunk 1]
```diff
@@ -1990,39 +1990,20 @@ struct Document {
                 ZoomOutIfNoGrid(dc);
                 return nullptr;
 
-            // FIXME: this functionality is really SCHOME, SHOME should be within line
-            case A_SHOME:
+            case A_SHOME: case A_SEND: case A_CHOME: case A_CEND: case A_HOME: case A_END: {
```
Comment (t2b3) [inline src/document.h:1993]: Is it admissible? I say this is not the code style, but it saves some lines...

**Comment by t2b3** [comment]
@aardappel 
Do you think the extra abstraction is justified in this case? Thanks in advance for looking into this.

**Comment by aardappel** [comment]
Yes, I guess this one is fine :)

**Comment by t2b3** [comment]
> Yes, I guess this one is fine :)

Thanks for having looked into this! Have a nice evening :-)
Best regards
Tobias

**Pull request merged**